sdk/src/account.rs

```diff
@@ -65,6 +65,9 @@ impl Account {
     }
 
     pub fn serialize_data<T: serde::Serialize>(&mut self, state: &T) -> Result<(), bincode::Error> {
+        if bincode::serialized_size(state)? > self.data.len() as u64 {
+            return Err(Box::new(bincode::ErrorKind::SizeLimit));
+        }
```

Comment:
So I was wondering about this. According to bincode's serialize, it shouldn't be an issue. Why did we conclude it was a problem again?

https://docs.rs/bincode/0.8.0/bincode/fn.serialize_into.html

Reply:
I thought you'd had a problem.  I'd had a problem: when I supplied 1 fewer byte than required for a vector of (u64, Hash).

Reply:
I'm looking at bincode now, and it looks like the check is in place if config has a limit in place, but the default bincode config is unlimited, so the check is skipped?

https://github.com/servo/bincode/blob/e84edd4b0ec2b4e5ebf70ea1508cce8db950fdcd/src/internal.rs#L20

https://github.com/servo/bincode/blob/e84edd4b0ec2b4e5ebf70ea1508cce8db950fdcd/src/config.rs#L98

Reply:
Yeah I remember having a problem with this but decided to just take another look at it and saw that documentation. But you're right, it seems we're responsible for setting the limit there.

Reply:
so two ways to do it, your way or pass in the account.data.len() as the size limit?

Reply:
bincode::config().limit(account.data.len() as u64).serialize_into() is uglier, IMHO